### DISCUSSION TITLE
Workspace Restore Request

### DISCUSSION THREAD

**Discussion by kacperpaczos** [linuxmint/discussions#49] [Ideas]
Linux Mint should restore the workspace after restarting the computer. Sometimes, I accidentally close the system and I would like to return to where I left off.

I think it would be possible to record user actions and information about the running applications, such as flatpak/snap/regular ones, and maintain a list where each newly launched application would be assigned an ID.
In the event of an unexpected system shutdown or when the user selects such an option, the previous workspace state could be restored.

We don't have control over external applications, but web browsers usually offer to restore tabs after an unexpected shutdown. Other programs can reload without any issues, like messaging apps, Discord, etc. You just need to open their respective windows.

The main task is to adapt system applications, such as xed, to remember unsaved files and restore tabs.

**Comment by anaximeno** [comment]
Something like hibernation?

**Reply by kacperpaczos** [reply]
Yes, unfortunately, hibernation ?is off? and I can't use them; I have 5 computers or laptops, and this option is disabled for me.

Another case: unexcited computer power off. MacOS restores 'session' of the user and brings back all windows. He's doing this without any information, so it's sometimes annoying. I believe that we should ask the user if they want to restore a session.

**Reply by anaximeno** [reply]
Having hibernation active by default in the regular Mint editions would be nice. It is actually (at least from what I tested) active in the LMDE 5 edition, and there's an option in the  Power Management setting module which lets you define the behavior of the system when the battery is critically low, there you could choose to just close the device or to hibernate it (when having hibernation active in the system). So it has the behavior you want, but it is just not active in the regular Ubuntu-based editions, for some reason.

**Comment by igorsantos07** [comment]
I don't really understand why hibernation is such a complex topic in Linux-land. I mean, Windows has that for what, 26 years? I believe it got introduced in Windows Me, yikes.

That said, I guess there are technical challenges and yadda-yadda. Thus, recording where each window was before closing then would be enough to help alleviating that pain. A LOT.
